packages/features/webhooks/lib/cron.ts

```diff
@@ -31,20 +31,35 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
         lte: dayjs().toISOString(),
       },
     },
+    select: {
+      id: true,
+      jobName: true,
+      payload: true,
+      subscriberUrl: true,
+      webhook: {
+        select: {
+          secret: true,
+        },
+      },
+    },
   });
 
+  const fetchPromises: Promise<any>[] = [];
+
   // run jobs
   for (const job of jobsToRun) {
     // Fetch the webhook configuration so that we can get the secret.
     const [appId, subscriberId] = job.jobName.split("_");
+    let webhook = job.webhook;
 
-    let webhook;
-    try {
-      webhook = await prisma.webhook.findUniqueOrThrow({
-        where: { id: subscriberId, appId: appId !== "null" ? appId : null },
-      });
-    } catch {
-      logger.error(`Error finding webhook for subscriberId: ${subscriberId}, appId: ${appId}`);
+    if (!webhook) {
+      try {
+        webhook = await prisma.webhook.findUniqueOrThrow({
```

Comment:
We now save the webhook to `webhookScheduledTriggers` so we don't have to do that anymore for each job